src/main/resources/META-INF/resources/primefaces/overlaypanel/overlaypanel.js

```diff
@@ -55,6 +55,7 @@ PrimeFaces.widget.OverlayPanel = PrimeFaces.widget.DynamicOverlayWidget.extend({
         this.cfg.hideEvent = this.cfg.hideEvent||'click.ui-overlaypanel';
         this.cfg.dismissable = (this.cfg.dismissable === false) ? false : true;
         this.cfg.showDelay = this.cfg.showDelay || 0;
+        this.appendTo = this.cfg.appendTo;
```

Comment:
i would do it like others here:
this.cfg.appendTo = PrimeFaces.utils.resolveAppendTo(...)

and remove the IF around removeDynamicOverlay
but please make sure that removeDynamicOverlay correclty skips "null" appendTo.
so we dont need always this If-statement in the widgets.

Reply:
I made the changes you requested and re-ran all the integration tests and things are still passing as expected after this change.